### PR TITLE
sec(componentTemplate): secure the page

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

Same as #9762 but for 2.8.x

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x

<h2> How this pull request can be tested ? </h2>

See Jira issue

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/views/componentTemplates/DB-Func.php [hunk 1]
```diff
@@ -39,26 +39,42 @@
...
+    $query = "SELECT compo_id FROM `giv_components_template` WHERE `name` = '" . CentreonUtils::escapeSecure($name) . "'";
+
+    if (!empty($formValues['host_id'])) {
+        if (preg_match('/([0-9]+)-([0-9]+)/', $formValues['host_id'], $matches)) {
+            $formValues['host_id'] = (int) $matches[1];
+            $formValues['service_id'] = (int) $matches[2];
+        } else {
+            throw new \InvalidArgumentException('host_id must be a combination of integers');
+        }
+    }
```
Comment (jeremyjaouen) [inline www/include/views/componentTemplates/DB-Func.php:59]: Why not using your method `parseHostIdPostParameter` ?

File: www/include/views/componentTemplates/DB-Func.php [hunk 2]
```diff
@@ -67,26 +83,39 @@ function NameHsrTestExistence($name = null)
...
+    $query = "SELECT compo_id FROM giv_components_template WHERE name = '" . CentreonUtils::escapeSecure($name) . "'";
+
+    if (!empty($formValues['host_id'])) {
+        if (preg_match('/([0-9]+)-([0-9]+)/', $formValues['host_id'], $matches)) {
+            $formValues['host_id'] = (int) $matches[1];
+            $formValues['service_id'] = (int) $matches[2];
+        } else {
+            throw new \InvalidArgumentException('host_id must be a combination of integers');
+        }
```
Comment (jeremyjaouen) [inline www/include/views/componentTemplates/DB-Func.php:99]: Why not using your method `parseHostIdPostParameter` ?

File: www/include/views/componentTemplates/DB-Func.php [hunk 3]
```diff
@@ -149,157 +179,195 @@ function multipleComponentTemplateInDB($compos = array(), $nbrDup = array())
...
-    if (isset($ret["host_id"]) && preg_match('/\d+\-\d+/', $ret["host_id"])) {
-        list($host_id, $service_id) = explode('-', $ret["host_id"]);
-        $rq .= "'" . $host_id . "', '" . $service_id . "', ";
+    if (!empty($formValues['host_id'])) {
+        if (preg_match('/([0-9]+)-([0-9]+)/', $formValues['host_id'], $matches)) {
+            $formValues['host_id'] = (int) $matches[1];
+            $formValues['service_id'] = (int) $matches[2];
+        } else {
+            throw new \InvalidArgumentException('chartId must be a combination of integers');
+        }
```
Comment (jeremyjaouen) [inline www/include/views/componentTemplates/DB-Func.php:215]: Why not using your method `parseHostIdPostParameter` ?

File: www/include/views/componentTemplates/DB-Func.php (hunk 2 above)
Comment (adr-mo) [reply www/include/views/componentTemplates/DB-Func.php:99]: Can't be applied for this case

**Review by jeremyjaouen — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**